.buildkite/common.py

```diff
@@ -174,35 +174,44 @@ def __call__(self, parser, namespace, value, option_string=None):
 )
 
 
-def revision_a():
-    """Determine if there is a base revision"""
-    if os.environ.get("BUILDKITE_PULL_REQUEST", "false") != "false":
-        return os.environ.get("BUILDKITE_PULL_REQUEST_BASE_BRANCH", "main")
-    return os.environ.get("REVISION_A")
-
-
 def random_str(k: int):
     """Generate a random string of hex characters."""
     return "".join(random.choices(string.hexdigits, k=k))
 
 
+def ab_revision_build(revision):
+    """Generate steps for building an A/B-test revision"""
+    # Copied from framework/ab_test. Double dollar signs needed for Buildkite (otherwise it will try to interpolate itself)
+    return [
+        f"commitish={revision}",
+        f"if ! git cat-file -t $$commitish; then commitish=origin/{revision}; fi",
+        "branch_name=tmp-$$commitish",
+        "git branch $$branch_name $$commitish",
+        f"git clone -b $$branch_name . build/{revision}",
+        f"cd build/{revision} && ./tools/devtool -y build --release && cd -",
+    ]
+
+
 def shared_build():
     """Helper function to make it simple to share a compilation artifacts for a
     whole Buildkite build
     """
-    build_cmds = ["./tools/devtool -y build --release"]
 
-    # If we are running in a PR context, then also build the base branch in the
-    # expected location, for the A/B tests machinery.
-    rev_a = revision_a()
+    # We need to support 3 scenarios here:
+    # 1. We are running in the nightly pipeline - only compile the HEAD of main.
+    # 2. We are running in a PR pipeline - compile HEAD of main as revision A and HEAD of PR branch as revision B.
+    # 3. We are running in an A/B-test pipeline - compile what is passed via REVISION_{A,B} environment variables.
```

Comment:
I think this is clearer. Thanks!

Reply:
Thanks! I'm honestly not particularly happy about the duplication of the checkout/clone logic and the implicit contract about naming of directories/branches between the build step and ab_test.py, but it'll have to do for now